File: client/src/app/modules/quotation/services/customer.order.status.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { CustomerOrderStatus } from '../model/CustomerOrderStatus';

@Injectable({
  providedIn: 'root'
})
export class CustomerOrderStatusService extends AppRestService<CustomerOrderStatus>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getCustomerOrderStatus() {
    return this.getListCached(new HttpParams(), "customer-order-status");
  }

  getCustomerStatusByCode(status: CustomerOrderStatus[], code: string) {
    if (status)
      for (let statu of status)
        if (statu.code == code)
          return statu;
    return null;
  }
}

File: src/main/java/com/jss/osiris/modules/quotation/service/CustomerOrderStatusService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.CustomerOrderStatus;

public interface CustomerOrderStatusService {
    public List<CustomerOrderStatus> getCustomerOrderStatus();

    public CustomerOrderStatus getCustomerOrderStatus(Integer id);

    public CustomerOrderStatus addOrUpdateCustomerOrderStatus(CustomerOrderStatus CustomerOrderStatus);

    public CustomerOrderStatus getCustomerOrderStatusByCode(String code);

    public void updateStatusReferential() throws Exception;

}
